src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_commands.c

```diff
@@ -234,3 +252,77 @@ static void wm_agent_upgrade_start_upgrades(cJSON *json_response, const cJSON* t
         mtwarn(WM_AGENT_UPGRADE_LOGTAG, WM_UPGRADE_NO_AGENTS_TO_UPGRADE);
     }
 }
+
+char* wm_agent_upgrade_process_agent_result_command(const int* agent_ids, wm_upgrade_agent_status_task* task) {
+    cJSON *response = cJSON_CreateArray();
+    cJSON *message_array = cJSON_CreateArray();
+    // Only one id of agent will reach at a time
+    int agent_id = agent_ids[0];
+    cJSON *message_object = wm_agent_upgrade_parse_task_module_request(WM_UPGRADE_AGENT_STATUS, agent_id);
+    cJSON_AddStringToObject(message_object, "status", task->status);
+    mtinfo(WM_AGENT_UPGRADE_LOGTAG, WM_UPGRADE_ACK_RECEIVED, agent_id, task->error_code, task->message);
+    cJSON_AddItemToArray(message_array, message_object);
+
+    // Send task update to task manager and bring back the response
+    wm_agent_upgrade_parse_task_module_task_ids(response, message_array);
```

Comment:
I think you need to create another function in the parser to do this and the validation of the function `wm_agent_upgrade_validate_task_update_message` all together.